### PR TITLE
CORE-6703 Merkletree specific KDoc updates

### PR DESCRIPTION
**Pull request by vlajos**
Merkle tree specific KDoc updates.

### CODE REVIEW

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt [hunk 1]
```diff
@@ -8,18 +8,46 @@ import net.corda.v5.crypto.SecureHash
  * [MerkleTreeHashDigestProvider]s let us specify the
  *  - Leaf Nonce
  *  - Leaf Hash
- *  - Node Hash calculation methods and
- *  - the base Digest Algorithm.
+ *  - Node Hash calculation methods
+ *  - Base Digest Algorithm.
```
Comment (nadinequinr3) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt]:
```suggestion
 *  - Base Digest Algorithm
```

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt [hunk 2]
```diff
@@ -8,18 +8,46 @@ import net.corda.v5.crypto.SecureHash
...
 
 interface MerkleTreeHashDigestProvider {
+    /**
+     * @property digestAlgorithmName Specifies the digest algorithm.
+     */
     val digestAlgorithmName: DigestAlgorithmName
 
+    /**
+     * Calculates the nonce for a leaf.
+     * @param index the leaf's index.
```
Comment (nadinequinr3) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt]:
```suggestion
     * @param index The leaf's index.
```

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt [hunk 3]
```diff
@@ -8,18 +8,46 @@ import net.corda.v5.crypto.SecureHash
...
+     */
     fun leafNonce(index: Int): ByteArray?
+
+    /**
+     * Calculates the hash of a leaf.
+     * @param index the leaf's index.
```
Comment (nadinequinr3) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt]:
```suggestion
     * @param index The leaf's index.
```

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt [hunk 4]
```diff
@@ -8,18 +8,46 @@ import net.corda.v5.crypto.SecureHash
...
+     * @param nonce the leaf's nonce.
```
Comment (nadinequinr3) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt]:
```suggestion
     * @param nonce The leaf's nonce.
```

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt [hunk 5]
```diff
@@ -8,18 +8,46 @@ import net.corda.v5.crypto.SecureHash
...
+     * @param bytes the leaf's content bytes.
```
Comment (nadinequinr3) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt]:
```suggestion
     * @param bytes The leaf's content bytes.
```

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt [hunk 6]
```diff
@@ -8,18 +8,46 @@ import net.corda.v5.crypto.SecureHash
...
     fun nodeHash(depth: Int, left: SecureHash, right: SecureHash): SecureHash
 }
 
+/**
+ * Special Digest provider for supporting size proofs.
+ */
 interface MerkleTreeHashDigestProviderWithSizeProofSupport : MerkleTreeHashDigestProvider {
+    /**
+     * Returns a size proof.
+     * @param leaves the tree's leaves.
```
Comment (nadinequinr3) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt]:
```suggestion
     * @param leaves The tree's leaves.
```

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTree.kt [hunk 7]
```diff
@@ -25,7 +25,7 @@ interface MerkleTree {
     val root: SecureHash
```
Comment (nadinequinr3) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTree.kt:25]: "@property leaves are the input data elements. Usually the hashes of something larger deterministically serialized."
(line 13)
Suggest:
The input data...

**Review by nadinequinr3 — CHANGES_REQUESTED** [review]
@vlajos I've added a couple of edits. From a Tech Writing perspective, everything else looks ok but it would be worth you reviewing again to make sure a CorDapp Developer has all of the information they would need to use any element. Thanks.

**Review by nadinequinr3 — DISMISSED** [review]
LGTM

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleProof.kt [hunk 8]
```diff
@@ -29,7 +29,7 @@ interface MerkleProof {
     val hashes: List<SecureHash>
 
     /**
-     * [verify] can be used to check if the [MerkleProof] has been generated from a [MerkleTree] with the given [root].
+     * Checks if the [MerkleProof] has been generated from a [MerkleTree] with the given [root].
```
Comment (mnesbit) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleProof.kt:32]: Should have @Param and @Returns info

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTree.kt [hunk 9]
```diff
@@ -10,7 +10,7 @@ import net.corda.v5.crypto.SecureHash
  */
 interface MerkleTree {
     /**
-     * @property leaves are the input data elements. Usually the hashes of something larger deterministically serialized.
+     * @property leaves The input data elements. Usually the hashes of something larger deterministically serialized.
```
Comment (mnesbit) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTree.kt]: In practice these are actually serialized objects not hashes in our current data model

File: crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt [hunk 10]
```diff
@@ -8,18 +8,46 @@ import net.corda.v5.crypto.SecureHash
...
+     */
     fun nodeHash(depth: Int, left: SecureHash, right: SecureHash): SecureHash
 }
 
+/**
+ * Special Digest provider for supporting size proofs.
+ */
 interface MerkleTreeHashDigestProviderWithSizeProofSupport : MerkleTreeHashDigestProvider {
+    /**
+     * Returns a size proof.
```
Comment (mnesbit) [inline crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestProvider.kt]: perhaps expand a bit.
Returns a size proof that reveals the number of leaves in the Merkle tree, but not the content of the leaves.

**Review by mnesbit — APPROVED** [review]
OK

**Pull request merged**